test/parallel/test-fs-write-reuse-callback.js

```diff
@@ -0,0 +1,38 @@
+// Flags: --expose-gc
+'use strict';
+const common = require('../common');
+const tmpdir = require('../common/tmpdir');
+const assert = require('assert');
+const path = require('path');
+
+// Regression test for https://github.com/nodejs/node-v0.x-archive/issues/814:
+// Make sure that Buffers passed to fs.write() are not garbage-collected
+// even when the callback is being reused.
+
+const fs = require('fs');
+
+tmpdir.refresh();
+const filename = path.join(tmpdir.path, 'test.txt');
+const fd = fs.openSync(filename, 'w');
+
+const size = 16 * 1024;
+const writes = 1000;
+let done = 0;
+
+const ondone = common.mustCall((err) => {
+  assert.ifError(err);
+  if (++done < writes) {
+    if (done % 25 === 0) global.gc();
+    setImmediate(write);
+  } else {
+    assert.strictEqual(
+      fs.readFileSync(filename, 'utf8'),
+      'x'.repeat(writes * size));
```

Comment:
The test is timing out on Raspberry Pi devices in CI. I suspect it is due to the known issue that (probably because they are operating on NFS mounts) opening a file on those devices can keep the event loop open if there is not an explicit close. I suspect this change or something similar will resolve that issue.

```suggestion
      'x'.repeat(writes * size));
    fs.closeSync(fd);
```

Reply:
Yeah, let’s just try that :)